backend/s3/file.go

```diff
@@ -209,11 +209,12 @@ func (f *File) Close() error {
 		if err != nil {
 			return err
 		}
+		return waitUntilFileExists(f, 5)
```

Comment:
we should check for the error and if return it, if any. But on success, we still need to call the line below where we set `f.writeBuffer = nil` because the buffer should be emptied any time we close() a file.